File: iam-policy-lens/scripts/gapic.py
from dataclasses import dataclass

@dataclass
class GapicCall:
    fullname: str
    file_path: str
    line: int
    source_line: str
    resolution: str

def clean_gapic_fullname(name: str) -> str:
    """Removes internal package structures like .services.[...].client."""
    import re
    name = re.sub(r'\.services\.[a-zA-Z0-9_]+\.client\.', '.', name)
    name = re.sub(r'\.client\.Client', '.Client', name)
    return name

File: iam-policy-lens/scripts/ts/gapic.ts
import { CredentialsInfo } from "./credentials.js";

const RELEVANT_PACKAGES = [
  "@google-cloud/",
  "@google/genai",
  "@google/adk",
  "googleapis",
  "@google-cloud/vertexai"
];

export interface GapicCall {
  fullname: string;
  file_path: string;
  line: number;
  source_line: string;
  resolution: string;
  credentials?: CredentialsInfo;
}

export function isRelevantPackage(pkgPath: string): boolean {
  return RELEVANT_PACKAGES.some(p => pkgPath.includes(p));
}
